packages/vanilla/src/components/masthead/masthead.template.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { getAttributes, toString } from '@carbon/icon-helpers';
+import close from '@carbon/icons/es/close/20';
+import cx from 'classnames';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import mastheadLeftnav from './mastheadLeftnav.template';
+import mastheadNav from './mastheadNav.template';
+import mastheadPlatform from './mastheadPlatform.template';
+import mastheadProfile from './mastheadProfile.template';
+import mastheadSearch from './mastheadSearch.template';
+import menu from '@carbon/icons/es/menu/20';
+import { settings } from 'carbon-components';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+/**
+ * renders ibm logo and masthead container
+ *
+ * @param {object} props Property elements passing into the template
+ * @param {boolean} props.hasProfile Determines whether to render Profile component
+ * @param {boolean} props.hasSearch Determines whether to render Search Bar
+ * @param {object} props.navigation Object containing navigation elements
+ * @param {object} props.profileData Object containing profile menu elements
+ * @returns {object} JSX object
+ */
+const mastheadTemplate = ({
+  navigation,
+  platform,
+  searchProps,
+  profileData,
+}) => {
+  const menuIcon = toString({
+    ...menu,
+    attrs: {
+      ...menu.attrs,
+      class: `${prefix}--side-nav__open`,
+    },
+  });
+  const closeIcon = toString({
+    ...close,
+    attrs: {
+      ...close.attrs,
+      class: `${prefix}--side-nav__close`,
+    },
+  });
+
+  const hasPlatform = cx({
+    [`${prefix}--masthead__platform`]: platform,
+  });
+
+  return `
+    <div class="${prefix}--masthead">
+      <div class="${prefix}--masthead__l0">
+        <header aria-label="IBM" data-autoid="${stablePrefix}--masthead" class="${prefix}--header" role="banner">
+          <a class="${prefix}--skip-to-content" href="#main-content" tabindex="0">Skip to main content</a>
```

Comment:
`Skip to main content` => do we need to have this internationalized?

Reply:
Yes, we actually mentioned this a while ago when we were doing the React masthead. Ideally this would come from the translation files and not another prop. We could add them in as props now since we have no real timing as to when new translation files will be available. We'd have to do it for the React component as well.